File: javastory.budgetsh.stage4/src/javastory/budgetsh/stage4/server/store/budget/TransactionStore.java
package javastory.budgetsh.stage4.server.store.budget;

import java.util.Collection;

import javastory.budgetsh.stage4.share.domain.budget.budget.Transaction;

public interface TransactionStore {
	//
	void remove(String id);
	void update(Transaction transaction);
	boolean exist(String id);
	Transaction retrieve(String id, String id2);
	boolean register(Transaction transaction);
	Collection<Transaction> retrieveAll(String id);

}

File: javastory.budgetsh.stage5.1/src/step3/store/filestore/TransactionStore.js
const TransactionFile = require('./io/TransactionFile.js');

exports.regist = function(transaction){
	TransactionFile.regist(transaction);
	return true;
}

exports.exist = function(id){
	if(TransactionFile.exist(id)){
		return true;
	}
	return false;
}

exports.retrieve = function(id, cashbookId){
	return TransactionFile.retrieve(id, cashbookId);
}

exports.retrieveAll = function(cashbookId){
	return TransactionFile.retrieveAll(cashbookId);
}

exports.update = function(transaction){
	TransactionFile.update(transaction);
}

exports.remove = function(id){
	TransactionFile.remove(id);
}